cassandra/pool.py

```diff
@@ -694,12 +694,14 @@ def _open_connection_to_missing_shard(self, shard_id):
                 return
         shard_aware_endpoint = self._get_shard_aware_endpoint()
         log.debug("shard_aware_endpoint=%r", shard_aware_endpoint)
-
         if shard_aware_endpoint:
-            conn = self._session.cluster.connection_factory(shard_aware_endpoint, on_orphaned_stream_released=self.on_orphaned_stream_released,
-                                                            shard_id=shard_id,
-                                                            total_shards=self.host.sharding_info.shards_count)
-            conn.original_endpoint = self.host.endpoint
+            try:
+                conn = self._session.cluster.connection_factory(shard_aware_endpoint, on_orphaned_stream_released=self.on_orphaned_stream_released,
+                                                                shard_id=shard_id,
+                                                                total_shards=self.host.sharding_info.shards_count)
+                conn.original_endpoint = self.host.endpoint
+            except Exception as exc:
+                log.error("Failed to open connection to %s, on shard_id=%i: %s", self.host, shard_id, exc)
```

Comment:
reraise the exception here, otherwise it gonna hide all possible failure from the connection creation, and there plently of those.

also I'm not sure how this fixes anything regarding the blocked shard aware port

Reply:
This PR doesn't fix the issue mentioned in the comment.

As mentioned in the issue, the log messages are very misleading - they don't suggest that shard-aware port is blocked. The only thing I change here is to log something in case of shard-aware CQL connection failure.

The main issue (`ThreadPoolExecutor` blocking enqueued tasks) is still there.